SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -153,6 +153,16 @@ void ActorSystem::ProcessUpdates(const FEntitySubViewUpdate& SubViewUpdate)
 				ApplyComponentAdd(Delta.EntityId, Change.ComponentId, Change.Data);
 				ComponentAdded(Delta.EntityId, Change.ComponentId, Change.Data, ToResolveOps);
 			}
+			for (const ComponentChange& Change : Delta.ComponentsRemoved)
+			{
+				// NOTE: We have to deal the with dormant before ComponentUpdated
+				// otherwise a SubViewUpdate that contains a component update that
+				// also removes the dormant component will have its updates ignored.
```

Comment:
I don't fully understand this what else can remove the dormant component? Can we do all component removes before updates or only this one?

Reply:
Basically the dormant component still being there causes the update to be ignored which normally is fine but is a race if both the dormant component are removed and updates occur.

Reply:
In ComponentUpdated we still have no channel but the ActorSubView also does not have the SpatialConstants::DORMANT_COMPONENT_ID component. We need ComponentRemoved to make a channel for us first.
```
	USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(EntityId);
	if (Channel == nullptr)
	{
		// If there is no actor channel as a result of the actor being dormant, then assume the actor is about to become active.
		if (ActorSubView->HasComponent(EntityId, SpatialConstants::DORMANT_COMPONENT_ID))
		{
			// ...
		}
		else
		{
			// <-- we hit this
			UE_LOG(LogActorSystem, Log,
				   TEXT("Worker: %s Entity: %lld Component: %d - No actor channel for update."));
			return;
		}
	}

Reply:
Lovely

Reply:
Can we assume that the existence of the update means that it's no longer dormant? Or that we create a channel for doing them?

Reply:
Thinking about how to make this do something easier I think it turns it into a bigger PR. It feels like a case of having the spatial and unreal stuff very intertwined in their assumptions. We should probably make dormancy a subview filter but I'm wondering if there is ever a case where you can receive updates on a dormant actor that isn't going to remain dormant? Or if there is ever a case where something is going to stop being dormant but receive no updates.

Reply:
Talked through this offline.